src/js/debug.js

```diff
@@ -0,0 +1,73 @@
+require.scopes.debug = (function() {
+let cookieFields = ['name', 'domain', 'hostOnly', 'path', 'secure', 'httpOnly', 'sameSite', 'session', 'expirationDate'];
+
+function printCookies(domain) {
+  getCookies(domain, true, (cookie) => {
+    console.log('----------------');
+    console.log(JSON.stringify(cookie, null, 2));
+  });
+}
+
+function getCookies(domain, showRelated, cookieCallback) {
+  let out ={};
+  if (showRelated) {
+    domain = window.getBaseDomain(domain);
+  }
+  out[domain] = [];
+  return new Promise((resolve) => {
```

Comment:
Generally? They help avoid excessive nestedness, which makes code more readable.

Also [some API's](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/webRequest) are able to receive promises and values interchangeably.